tests/test_internals/test_movie_utils.py

```diff
@@ -110,3 +115,99 @@ def test_isopened_assertions(self):
         self.assertRaises(AssertionError, vcc.get_movie_frame_count)
         self.assertRaises(AssertionError, vcc.get_movie_frame_dtype)
         self.assertRaises(AssertionError, vcc.__next__)
+
+
+@unittest.skipIf(not CV2_INSTALLED, "cv2 not installed")
+class TestMovieInterface(unittest.TestCase):
+
+    frame_shape = (800, 600, 3)
+    number_of_frames = 500
+    fps = 25
+
+    def setUp(self):
+        self.test_dir = tempfile.mkdtemp()
+        self.nwbfile = make_nwbfile_from_metadata(dict(NWBFile=dict(session_start_time=datetime.now())))
+        self.nwbfile_path = os.path.join(self.test_dir, "movie_test.nwb")
+
+    def create_movie(self, fps, frame_shape, number_of_frames):
+        movie_frames = np.random.randint(0, 255, size=[number_of_frames, *frame_shape], dtype="uint8")
+        movie_file = os.path.join(self.test_dir, "test.avi")
+        writer = cv2.VideoWriter(
+            filename=movie_file,
+            apiPreference=None,
+            fourcc=cv2.VideoWriter_fourcc(*"HFYU"),
+            fps=fps,
+            frameSize=frame_shape[1::-1],
+            params=None,
+        )
+        for k in range(number_of_frames):
+            writer.write(movie_frames[k, :, :, :])
+        writer.release()
+        return movie_file
+
+    def test_iterator_general(self):
+        movie_file = self.create_movie(self.fps, self.frame_shape, self.number_of_frames)
+        it = H5DataIO(MovieDataChunkIterator(movie_file), compression="gzip")
```

Comment:
```suggestion
        it = H5DataIO(MovieDataChunkIterator(movie_file))
```
I don't think we need to use compression when running the tests here, probably just makes them take longer. Chunking should be handled automatically though within the `H5DataIO`

Reply:
Can also be changed in the rest of the testing file.

Reply:
do we not want to test the fact that given the custom `MovieDataChunkIterator`, `H5DataIO` is able to compress + write data correctly so that it can be read. 
I mean, is there a possibility if things run fine without compression but not with.

Reply:
That would be the responsibility of unit tests of the `H5DataIO` on the HDMF side, which is very thoroughly done. Our only job is to make sure the data itself reads and writes through the IO (namely the interaction with chunking).

One time we do test compression setting is on the Interface level, where we are passing it as a higher-level input argument on our own side. But that's still not really for testing that the actual compression works, just testing that the call gets passed down.